### ISSUE TITLE
[💡Feature Request]

### ISSUE THREAD

**Issue by shershah010** [tfukaza/harvest#123]
**Describe the feature**
Software like splunk or xsoar have local servers with guis fro users to interface with there system. These servers also make it easier for distributed environments and the ability to run inside a docker container while being able to easily access it outside the container. I think if Harvest had a similar feature, it would be pretty sweet.

I think that this feature is pretty big and probably won't fit into the second milestone, but could be a good third or fourth milestone.

Here are some mockups of how the web app would look:

file:///home/sher/Downloads/Harvest.zip

**Comment by tfukaza** [comment]
It's probably better to iterate GUI stuff on Figma: https://www.figma.com/file/b6BfADl65XBriu5W1gQGN3/Harvest?node-id=0%3A1. 

As for my thoughts, we should abstract the inner workings of Harvest and only show users high-level actionable stuff. An analogy would be an AC unit where you control temperature by adjusting the voltage to the compressor pump. For the end users, it would be better to simply have a temperature control knob, and hide the inner workings of the AC. For example, the Storage class probably doesn't need its own tab, since once the algo is up and running, users don't really care if their data is being saving in the RAM or a pickle file. 

In other words, the UI should be designed based on _what the user wants to do_, not on _how the program is designed_.

**Comment by shershah010** [comment]
Yeah, I have the graphics here: https://www.figma.com/file/b6BfADl65XBriu5W1gQGN3/Harvest?node-id=132%3A2

I agree that the ui should be designed around what the users want to do, but specifying the storage type is something that a user would care about. If they want to save data in a database or as csv files, then they should be able to specify what format and where that data should be saved.

**Comment by shershah010** [comment]
Current work done on this branch: https://github.com/tfukaza/harvest/tree/gui

**Comment by tfukaza** [comment]
IMO we should only implement a minimal GUI until we agree on the final UI.
